server/resp/src/test/java/org/infinispan/server/resp/hll/HyperLogLogTest.java

```diff
@@ -0,0 +1,48 @@
+package org.infinispan.server.resp.hll;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.nio.charset.StandardCharsets;
+
+import org.infinispan.server.resp.hll.internal.CompactSet;
+import org.infinispan.server.resp.hll.internal.ExplicitSet;
+import org.testng.annotations.Test;
+
+@Test(groups = "functional", testName = "server.resp.hll.HyperLogLogTest")
+public class HyperLogLogTest {
+
+   public void testUnderlyingSetUpdates() {
```

Comment:
i've just given a real quick view, as I don't really understand (yet) what this is for

Reply:
Sure, np. The HyperLogLog class has two approaches to tracking the elements. There is an explicit and compact format. We always start with explicit, which keeps the element's hash in a HashSet. But we only use it until a certain point. More precisely, we count up to 192 with the explicit, and after that, it would use more space in memory than the compact, so we migrate the representation.

This test verifies that we properly migrate everything from an explicit to a compact format.